backend/file-orchestrator-service/api/src/commonMain/kotlin/PathUtils.kt

```diff
@@ -31,6 +31,8 @@ fun String.components(): List<String> = removePrefix("/").removeSuffix("/").spli
 
 fun String.fileName(): String = substringAfterLast('/')
 
+fun String.depth(): Int = split("/").size - 1
```

Comment:
I don't know if it will ever be the case, but won't this be wrong for paths like `/foo/`?

Something like
```
fun String.depth(): Int = split("/").filter {it.size > 0}.size - 1
```
Would fix this, but I'm not sure that would ever be the case.

Reply:
Actually, no. My case won't work either with that code snippet.

Reply:
based on a normalized path this is correct . Have however made an if on (equals("/")) since this should be depth 0 according to doc